yt_dlp/extractor/twitter.py

```diff
@@ -503,90 +607,114 @@ def add_thumbnail(name, size):
+                return {
+                    '_type': 'url',
+                    'url': get_binding_value('broadcast_url'),
+                    'ie_key': TwitterBroadcastIE.ie_key(),
+                }
+            elif card_name == 'summary':
+                return {
+                    '_type': 'url',
+                    'url': get_binding_value('card_url'),
+                }
+            elif card_name == 'unified_card':
+                media_entities = self._parse_json(get_binding_value('unified_card'), twid)['media_entities']
+                media = traverse_obj(media_entities, ..., expected_type=dict, get_all=False)
+                return extract_from_video_info(media)
+            # amplify, promo_video_website, promo_video_convo, appplayer,
+            # video_direct_message, poll2choice_video, poll3choice_video,
+            # poll4choice_video, ...
+            else:
+                is_amplify = card_name == 'amplify'
+                vmap_url = get_binding_value('amplify_url_vmap') if is_amplify else get_binding_value('player_stream_url')
+                content_id = get_binding_value('%s_content_id' % (card_name if is_amplify else 'player'))
+                formats, subtitles = self._extract_formats_from_vmap_url(vmap_url, content_id or twid)
+                self._sort_formats(formats)
+
+                thumbnails = []
+                for suffix in ('_small', '', '_large', '_x_large', '_original'):
+                    image = get_binding_value('player_image' + suffix) or {}
+                    image_url = image.get('url')
+                    if not image_url or '/player-placeholder' in image_url:
+                        continue
+                    thumbnails.append({
+                        'id': suffix[1:] if suffix else 'medium',
+                        'url': image_url,
+                        'width': int_or_none(image.get('width')),
+                        'height': int_or_none(image.get('height')),
+                    })
+
+                return {
+                    'formats': formats,
+                    'subtitles': subtitles,
+                    'thumbnails': thumbnails,
+                    'duration': int_or_none(get_binding_value(
+                        'content_duration_seconds')),
+                }
+
+        entries = []
+        media_path = ((None, 'quoted_status'), 'extended_entities', 'media', ...)
+        medias = traverse_obj(status, media_path, expected_type=dict, default=[])
+        for media in medias:
+            if not media or media.get('type') == 'photo':
+                continue
+
+            data = extract_from_video_info(media)
+            if data:
+                entries.append({**info, **data, 'display_id': twid})
+            continue
+
+        card = status.get('card')
+        if card:
+            data = extract_from_card_info(card)
+            if data:
+                entries.append({**info, **data, 'display_id': twid})
```

Comment:
Not a necessary change, but I think it's better to make `extract_from_card_info` accept an input of `None` and then do:
```suggestion
        data = extract_from_card_info(status.get('card'))
        if data:
            entries.append({**info, **data, 'display_id': twid})
```

Reply:
In the original extractor the logic went like this:

* look for anything in `status['extended_entities']['media'][0]`, extract from that
* otherwise look for status['card'], extract whatever from there
* otherwise try status['entities']['urls'][0]['expanded_url']

The final case wouldn't have run for any truthy value of `status.get('card')`.

Reply:
Do we need to mirror old behavior?

Since we extract multiple elements anyways, why not also use `card` extraction whenever possible?

From what I can see in the [docs](https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/entities#urls) that final case only makes sense if we want to follow redirects on no found media. Personally I would not want such behavior at all. In the case of scrapping it we could just `ExtractorError` immediately.